plugins/plugin-chart-table/src/types.ts

```diff
@@ -0,0 +1,87 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { QueryFormDataMetric } from '@superset-ui/query';
+import { ChartProps, DataRecord, DataRecordFilters } from '@superset-ui/chart';
+import { TimeFormatter, TimeGranularity } from '@superset-ui/time-format';
+import { NumberFormatter } from '@superset-ui/number-format';
+
+export enum DataType {
+  Number = 'number',
+  String = 'string',
+  DateTime = 'datetime',
+}
+
+export interface DataColumnMeta {
+  // `key` is what is called `label` in the input props
+  key: string;
+  // `label` is verbose column name used for rendering
+  label: string;
+  isTime?: boolean;
```

Comment:
If this is only for formatting, there should not be difference between `'metric'` and `'percent-metric'`? because the formatter can be specified by the field `formatter` ?

Reply:
Makes sense, will update.

Reply:
I removed these properties, and changed to use `dataType` instead. It now assumes `isMetric` when `dataType === 'number'`, which is a safe assumption at least for now.